pkg/webhook/patch.go

```diff
@@ -292,16 +293,78 @@ func addEnvironmentVariable(pod *corev1.Pod, envName, envValue string) *patchOpe
+	  2. Change VolumeMount mountPath for `spark.properties` to subPath
+	  3. subPath all other keys from the sparkConfMap spec provided ConfigMap
+	---
+	Edge cases:
+	  * Handling of duplicate `spark.properties` files which carry the configuration for the driver and executor(s)
+*/
+func addSparkConfigMap(pod *corev1.Pod, app *v1beta2.SparkApplication, client kubernetes.Interface) []patchOperation {
 	var patchOps []patchOperation
 	sparkConfigMapName := app.Spec.SparkConfigMap
 	if sparkConfigMapName != nil {
+		// Add Volume populated by the ConfigMap
 		patchOps = append(patchOps, addConfigMapVolume(pod, *sparkConfigMapName, config.SparkConfigMapVolumeName))
-		vmPatchOp := addConfigMapVolumeMount(pod, config.SparkConfigMapVolumeName, config.DefaultSparkConfDir)
-		if vmPatchOp == nil {
-			return nil
+		containerIndex := findContainer(pod)
+		// Note: no error handling is done for `containerIndex == -1` because we anticipate the Pod to handle the Container lifecycle
+		glog.V(3).Infof("SubPath patching existing VolumeMounts: %v", pod.Spec.Containers[containerIndex].VolumeMounts)
+		volumeMountIndex := findVolumeMountIndex(&pod.Spec.Containers[containerIndex])
+		if volumeMountIndex >= 0 {
+			// Change existing VolumeMount to exclusively subPath `spark.properties`
+			glog.V(2).Infof("Replacing VolumeMount %s at %v", pod.Spec.Containers[containerIndex].VolumeMounts[volumeMountIndex].Name, volumeMountIndex)
+			mnt := pod.Spec.Containers[containerIndex].VolumeMounts[volumeMountIndex]
+			mnt.MountPath = fmt.Sprintf("%s/%s", mnt.MountPath, config.DefaultSparkPropertiesFile)
+			mnt.SubPath = config.DefaultSparkPropertiesFile
+			replaceMountPath := fmt.Sprintf("/spec/containers/%d/volumeMounts/%d", containerIndex, volumeMountIndex)
+			glog.V(2).Infof("New VolumeMount %v", replaceMountPath)
+			patchOps = append(patchOps, patchOperation{Op: "replace", Path: replaceMountPath, Value: mnt})
+		} else {
+			glog.V(2).Infof("No pre-existing VolumeMount found. Not changing VolumeMount to SubPath.")
+			// Add VolumeMount to `config.DefaultSparkConfDir` (= /opt/spark/conf) compliant with Spark 2.4.x - 3.2.1
+			// Note: no assumption being made that spark generates a ConfigMap here, hence the VolumeMount mountPath fallback.
+			//       https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/216#issuecomment-698455388 will never
+			//       work because `spark.properties` has higher precedence than `spark-defaults.conf` anyways. We hence do not
+			//       need to subPath mount in the default case.
+			vmPatchOp := addConfigMapVolumeMount(pod, config.SparkConfigMapVolumeName, config.DefaultSparkConfDir)
+			if vmPatchOp == nil {
+				return nil
+			}
+			patchOps = append(patchOps, *vmPatchOp)
 		}
-		patchOps = append(patchOps, *vmPatchOp)
+		// Patch custom sparkConfigMap from spec to have them subPath mounted
+		configMap, err := client.CoreV1().ConfigMaps(app.Namespace).Get(context.TODO(), *sparkConfigMapName, metav1.GetOptions{})
```

Comment:
`if err == nil ` I think

Reply:
Thanks for the review. The `if err == nil` is not required here because we the `else` clause here:

https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1568/files/7cf4c30947ecb2db0fdc732b42a8e10c929361a1#diff-0aa888a76ed441948985bc01c47945dc5572d3173b19e7beef96d7947920b763R364-R366

Reply:
I understood that we patch when we found the configmap with NO error, so I think the correct condition is`if err == nil {
//patch
} else {
			glog.Errorf("Could not get custom spark config map: %v", err)
		}` 
or `if configMap != nil {
//patch
} else {
			glog.Errorf("Could not get custom spark config map: %v", err)
		}`

Reply:
Hi, we're currently waiting for a response from maintainers on inclusion. I would appreciate waiting for their response before opening CRs, especially for cosmetics. I won't be replying to ANY non maintainer reviews until a response is given, so most items such as this one will be moot until that happens. Thanks.

Reply:
@maknihamdi  is right ... we should patch the configmap only if there is **NO ERROR**, otherwise the current patch will always branch to else complaining that the configmap was not found... (tested this locally and it actually fails if err != nil)

Reply:
Hi @mrendi29, I'm sorry for not being clearer in my earlier response.
I wasn't trying to imply @maknihamdi is incorrect - I mentioned that due to a bit of history around this issue and to emphasize that investing any CR time here is irrelevant until the maintainers actually want to push this ahead. 
 
The original code ( and my own version ) has proper error handling https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/504c88e2003f32114981baea99cbbd1db1e82794/pkg/webhook/patch.go#L319C7-L319C7, so I'm guessing this regression happened during the last iteration.

I really don't mind updating the PR, but after two years I'm really unsure of the chances of it getting merged.